### DISCUSSION TITLE
EOF occurred in violation of protocol (_ssl.c:2393)

### DISCUSSION THREAD

**Discussion by AlRodriguezGar14** [aws/aws-cli#7735] [Q&A] [answered]
Occasionally, as this is a random issue, I get the `EOF occurred in violation of protocol (_ssl.c:2393)` error when trying to upload some content to the server with:
`aws --profile production s3 sync . s3://url`
Then the app can’t connect to the endpoint and crashes, stopping the upload.

As I said, this is quite random because I usually don’t have this issue.

OS: macOS Ventura 13.2.1
I installed aws-cli with homebrew. The current version is aws-cli/2.10.3
Python 3.11
Region configured is us-east-1

Additional config:
    max_concurrent_requests = 30
    max_bandwidth = 10GB/s
    multipart_chunksize = 100MB

**Comment by RyanFitzSimmonsAK** [comment]
Hi @AlRodriguezGar14, thanks for reaching out.  You mentioned this issue doesn't happen consistently; could you provide debug logs for both a successful sync and a failed one?  You can get debug logs by adding --debug to your command, and redacting any sensitive information.  Thanks!

**Reply by AlRodriguezGar14** [reply]
Thank you for the quick reply. Now I can't try it (hopefully tomorrow). A couple of hours ago, I did another test (without the --debug command, my mistake) and the output error was:

`aws --profile production s3 sync . s3://my-url/directory/
`
upload failed: path-to-the-file-I-want-to-upload/  to s3://my-url/directory/ Connection was closed before we received a valid response from endpoint URL: "https://my-url.s3.amazonaws.com/directory/file-i-was-uploading&partNumber=326"


Sorry If I'm not too clear and/or if I sound a bit lost, as I’m still a newbie on this. Hopefully, tomorrow I can share the proper logs 🙏

**Reply by RyanFitzSimmonsAK** [reply]
Hi @AlRodriguezGar14, are you still having this problem?  Thanks!

**Reply by AlRodriguezGar14** [reply, marked as answer]
> Hi @AlRodriguezGar14, are you still having this problem? Thanks!

Hi Ryan, the error didn’t repeat, which makes me think that there was some kind of a problem with AWS or the backend. I’m uploading large files, but everything seems to be working (the error appeared suddenly and it was quite random), so I can’t share with you a comparison between the successful uploads and canceled ones.

**Reply by RyanFitzSimmonsAK** [reply]
Glad to hear the issue is resolved.  If it begins happening once more, feel free to bring it up again.

**Comment by Satyaexp** [comment]
facing a similar issue   this is the error ssl.SSLEOFError: [SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1007)" DDuring handling of the above exception, another exception occurred: urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='bucketname.s3-accelerate.amazonaws.com', port=443): Max retries exceeded with url: /my_path/8fa39771.png?AWSAccessKeyId=AWSAccessKeyId&Signature=Signature&Expires=1715552122 (Caused by SSLError(SSLEOFError(8, '[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1007)')))

**Comment by gongshoudao** [comment]
same issue:
[SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1000

**Comment by nodakai** [comment]
When I encountered this issue on a server within a private LAN which required an [HTTP proxy](https://tinyproxy.github.io/) for AWS access, setting `aws configure set default.s3.max_concurrent_requests 1` effectively mitigated the problem.

* I'm not certain if limiting it to 1 was strictly necessary, maybe 2 or 3 could have sufficed
* This change might have come at the cost of reduced performance
